src/frequenz/sdk/power/_distribution_algorithm.py

```diff
@@ -396,24 +396,26 @@ def _greedy_distribute_remaining_power(
         Returns:
             Return the power for each inverter in given distribution.
         """
-        if remaining_power == 0:
+        if isclose(remaining_power, 0.0, abs_tol=1e-9):
             return DistributionResult(distribution, remaining_power)
 
-        new_distribution: Dict[int, int] = {}
+        new_distribution: Dict[int, float] = {}
 
         for inverter_id, power in distribution.items():
-            if remaining_power == 0 or power == 0:
+            if isclose(remaining_power, 0.0, abs_tol=1e-9) or isclose(
```

Comment:
Maybe have a wrapper with these parameters so you have to write them and document them only once?

You can even use [`functools.partial`](https://docs.python.org/3/library/functools.html#functools.partial) instead of an explicit wrapper function.

```py
# We use this tolerance because X and Y
is_close_to_zero = partial(math.isclose, b=0.0, abs_tol=1e-9)

if is_close_to_zero(remaining_power) or is_close_to_zero(power):
    ...
```

Reply:
This is great idea, I've implemented as suggested.